From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: MichaelSimons <msimons@microsoft.com>
Date: Wed, 24 Aug 2022 12:21:13 +0000
Subject: [PATCH] Lift System.Memory version

System.Memory version needs to align with the runtime since System.Collections.Immutable is lifted in order to prevent the following situation:
error NU1605: Detected package downgrade: System.Memory from 4.5.5 to 4.5.4. Reference the package directly from the project to select a different version.  [/repos/tarball-rc1/src/razor-compiler/artifacts/source-build/self/src/razor-compiler.sln]
error NU1605:  Microsoft.CodeAnalysis.Razor -> Microsoft.CodeAnalysis.CSharp 4.4.0 -> Microsoft.CodeAnalysis.Common 4.4.0 -> System.Collections.Immutable 7.0.0-rc.1.22422.23 -> System.Memory (>= 4.5.5)  [/repos/tarball-rc1/src/razor-compiler/artifacts/source-build/self/src/razor-compiler.sln]
error NU1605:  Microsoft.CodeAnalysis.Razor -> Microsoft.CodeAnalysis.CSharp 4.4.0 -> Microsoft.CodeAnalysis.Common 4.4.0 -> System.Memory (>= 4.5.4) [/repos/tarball-rc1/src/razor-compiler/artifacts/source-build/self/src/razor-compiler.sln]

Requires https://github.com/dotnet/source-build/issues/2482 in order to eliminate this patch
---
 eng/Versions.props | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/eng/Versions.props b/eng/Versions.props
index 627f9c46824..711593e2a1e 100644
--- a/eng/Versions.props
+++ b/eng/Versions.props
@@ -214,7 +214,7 @@
     <SystemIOPipesAccessControlVersion>5.0.0</SystemIOPipesAccessControlVersion>
     <SystemIOPipelinesVersion>6.0.1</SystemIOPipelinesVersion>
     <SystemManagementVersion>5.0.0-preview.8.20407.11</SystemManagementVersion>
-    <SystemMemoryVersion>4.5.4</SystemMemoryVersion>
+    <SystemMemoryVersion>4.5.5</SystemMemoryVersion>
     <SystemResourcesExtensionsVersion>6.0.0</SystemResourcesExtensionsVersion>
     <SystemRuntimeCompilerServicesUnsafeVersion>6.0.0</SystemRuntimeCompilerServicesUnsafeVersion>
     <SystemRuntimeInteropServicesNFloatInternalVersion>6.0.1</SystemRuntimeInteropServicesNFloatInternalVersion>
